packages/apps/shopify-api/runtime/http/headers.ts

```diff
@@ -1,15 +1,29 @@
 import type {Headers} from './types';
 
+/**
+ * Canonicalizes a header name by capitalizing each segment and ensuring consistent hyphenation.
+ *
+ * @param hdr - The header name to canonicalize.
+ * @returns The canonicalized header name.
+ */
 export function canonicalizeHeaderName(hdr: string): string {
-  return hdr.replace(
-    /(^|-)(\w+)/g,
-    (_fullMatch, start, letters) =>
-      start +
-      letters.slice(0, 1).toUpperCase() +
-      letters.slice(1).toLowerCase(),
-  );
+  return hdr
+    .split('-')
+    .map(
+      (segment) =>
+        segment.slice(0, 1).toLocaleUpperCase() +
+        segment.slice(1).toLocaleLowerCase(),
+    )
+    .join('-');
```

Comment:
I don't think we should do this, as I assume the regex version will be slightly more efficient.

Reply:
Yes we can revert, this is a more readable approach :)